File: workshop-materials/scripts/do_addition.R
do_addition <- function(number1, number2){
        
        # Check that arguments provided are numberic
        if(class(number1) != "numeric" | class(number2) != "numeric"){
                print("Error: one or more of your inputs are not numeric")
                
        # Perform simple addition and return the result
        } else{
                result <- number1 + number2
                return(result)
        }
}

File: my-reproducible-manuscript/scripts/do_addition.py
def addnrs(number1, number2):
    # Check that arguments provided are numeric
    if not isinstance(number1, (int, float)) or not isinstance(number2, (int, float)):
        print("Error: one or more of your inputs are not numeric")
    else:
        result = number1 + number2
        return result
